sklearn/base.py

```diff
@@ -398,6 +416,54 @@ def __sklearn_tags__(self):
             classifier_tags=None,
         )
 
+    # TODO(1.7): Remove this method
+    def _more_tags(self):
+        warnings.warn(
+            "The `_more_tags` method is deprecated in 1.6 and will be removed in "
+            "1.7. Please implement the `__sklearn_tags__` method.",
+            category=FutureWarning,
+        )
+        return _to_old_tags(default_tags(self))
+
+    # TODO(1.7): Remove this method
+    def _get_tags(self):
+        warnings.warn(
+            "The `_get_tags` tag provider is deprecated in 1.6 and will be removed in "
+            "1.7. Please implement the `__sklearn_tags__` method.",
+            category=FutureWarning,
+        )
+        # In case a user called `_get_tags` but that the estimator already did the job
+        # implementing `__sklearn_tags__` completely and removed `_more_tags`, let's
+        # default back to the future behaviour. Otherwise, we will get the default tags.
+        from sklearn.utils._tags import _find_tags_provider, _to_old_tags, get_tags
+
+        if _find_tags_provider(self, warn=False) == "__sklearn_tags__":
+            return _to_old_tags(get_tags(self))
+
+        collected_tags = {}
+        for base_class in reversed(inspect.getmro(self.__class__)):
+            if hasattr(base_class, "_more_tags"):
```

Comment:
This is the old implementation of `_get_tags`. So I would not touch it.
The only change here is to fallback to `__sklearn_tags__` in case someone in the chain of inheritance implemented it (like scikit-learn implementing the new tags and removing the old tags).